employee-rostering-backend/pom.xml

```diff
@@ -74,71 +84,31 @@
     </dependency>
     <dependency>
       <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-websocket</artifactId>
+      <artifactId>spring-boot-configuration-processor</artifactId>
+      <version>2.1.6.RELEASE</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-core</artifactId>
     </dependency>
+
     <!-- Testing dependencies -->
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-test</artifactId>
       <scope>test</scope>
-      <exclusions>
-        <exclusion>
-          <groupId>junit</groupId>
-          <artifactId>junit</artifactId>
-        </exclusion>
-      </exclusions>
     </dependency>
     <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter-api</artifactId>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <version>1.2.3</version>
```

Comment:
This is unnecessary too. Logback is provided by Spring Boot Starters. See [Logging](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-logging.html). You can verify it by running `mvn dependency:tree`, you'll see the artifact in the dependency tree.

Although it *is* your runtime dependency and it is correct to declare it under normal condition (we ban transitive dependencies in every other kiegroup project), it's against the "philosophy" of Spring Boot [Starters](https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-build-systems.html#using-boot-starter) which are purposefully used to drag in a bunch of dependencies in a consistent ways so that you don't have to declare them individually.

As an example Spring Boot app, I think we should follow this convention and not override the dependencies provided by Starters unless necessary.